autoload/ale/engine.vim

```diff
@@ -201,6 +201,16 @@ function! s:FixLocList(buffer, loclist) abort
     let l:last_line_number = ale#util#GetLineCount(a:buffer)
 
     for l:item in a:loclist
+        if g:ale_experimental_multibuffer
+            let l:bufnr = a:buffer
+
+            if has_key(l:item, 'bufnr') && l:item.bufnr >= 0
+              let l:bufnr = l:item.bufnr
+            endif
+
+            let l:last_line_number = ale#util#GetLineCount(l:bufnr)
```

Comment:
You should include a check here which skips over any items from other buffers `if !g:ale_experimental_multibuffer`. Then you don't have to worry about checking that in any of the linters.